src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,125 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Semantics;
+using Bicep.Core.Text;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // This handler is used to generate compiled .json file for given a bicep file path.
+    // It returns build succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepBuildCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager CompilationManager;
+
+        public BicepBuildCommandHandler(ICompilationManager compilationManager, ISerializer serializer)
+            : base(LanguageConstants.Build, serializer)
+        {
+            CompilationManager = compilationManager;
+        }
+
+        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string buildOutput = GenerateCompiledFileAndReturnBuildOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(buildOutput);
+        }
+
+        private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            string? compiledFilePath = GetCompiledFilePath(bicepFilePath, sb, out string bicepFile, out string compiledFile);
+
+            if (string.IsNullOrWhiteSpace(compiledFilePath))
+            {
+                throw new ArgumentException($"Invalid input file");
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                AppendDiagnosticsMessage(diagnosticsByFile, bicepFile, sb);
+                return sb.ToString();
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite))
```

Comment:
>FileShare.ReadWrite [](http://example.com/codeflow?start=115&length=19)

`FileShare.ReadWrite` means that some other process is allowed to read the file while we're writing, which probably won't do anything useful. We should probably just use the default `FileShare` setting for now.

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/daa24e89c632c399b92f0de41c98fcea77b930eb